pkg/solana/logpoller/log_poller.go

```diff
@@ -0,0 +1,316 @@
+package logpoller
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"iter"
+	"slices"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+)
+
+var (
+	ErrFilterNameConflict = errors.New("filter with such name already exists")
+)
+
+type ORM interface {
+	InsertFilter(ctx context.Context, filter Filter) (id int64, err error)
+	SelectFilters(ctx context.Context) ([]Filter, error)
+	DeleteFilters(ctx context.Context, filters []Filter) error
+	MarkFilterDeleted(ctx context.Context, id int64) (err error)
+}
+
+type LogPoller struct {
+	services.StateMachine
+	lggr logger.SugaredLogger
+	orm  ORM
+
+	filtersByName     map[string]Filter
+	filtersByAddress  map[PublicKey]map[EventSignature][]Filter
+	filtersToBackfill []Filter
+	filtersToDelete   []Filter // populated on start from db and pruned on first iteration of run
```

Comment:
This may not matter since I don't expect the number of filters to be registered to be too huge, but I notice there are at least 2 different copies of the same `Filter` struct that get saved in memory and possibly 3 or 4 in some cases.

We could reduce memory footprint by making these slices of `*Filter` instead of slices of `Filter` so there is only one copy around. (See elsewhere for a different suggestion about making the slices into maps; this could apply to either.) 

(Probably not worth worrying about for now.)

Reply:
IMHO, using pointers decreases readability as it gives the impression that the filter might be modified.
Unless we expect IDL to bea  large string,g I'd prefer to use struct instead of pointer.